internal/kubernetes/cert.go

```diff
@@ -0,0 +1,51 @@
+package kubernetes
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/aws/eks-hybrid/internal/api"
+	"github.com/aws/eks-hybrid/internal/node/hybrid"
+	"github.com/aws/eks-hybrid/internal/validation"
+)
+
+const KubeletServerCertPath = "/var/lib/kubelet/pki/kubelet-server-current.pem"
+
+type KubeletCertificateValidator struct {
+	installRoot string
+	logger      *zap.Logger
+}
+
+func WithInstallRoot(installRoot string) func(*KubeletCertificateValidator) {
+	return func(v *KubeletCertificateValidator) {
+		v.installRoot = installRoot
+	}
+}
+
+func NewKubeletCertificateValidator(logger *zap.Logger, opts ...func(*KubeletCertificateValidator)) KubeletCertificateValidator {
+	v := &KubeletCertificateValidator{
+		installRoot: KubeletServerCertPath,
+		logger:      logger,
+	}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return *v
+}
+
+func (v KubeletCertificateValidator) Run(ctx context.Context, informer validation.Informer, node *api.NodeConfig) error {
+	name := "kubernetes-kubelet-certificate"
+	var remedationErr error
+	informer.Starting(ctx, name, "Validating kubelet server certificate")
+	defer func() {
+		informer.Done(ctx, name, remedationErr)
+	}()
+
+	if err := hybrid.ValidateKubeletCert(v.logger, v.installRoot, node.Spec.Cluster.CertificateAuthority); err != nil {
+		remedationErr = validation.WithRemediation(err, err.Remediation)
```

Comment:
instead of doing this I would just make `ValidationError` implement the `Remediable` interface